Architecture.md

```diff
@@ -12,7 +12,11 @@
 
 #### Web application
 
-***TODO: [CLIENT] fill for web application frontend***
+The website is made using [React](https://react.dev/), we decided to use this framework because it is the most popular, widely used in industry and greatly documented. We also chose React since we are all familiar with it due to several past projects.
+
+As for the component library, we chose [Chakra UI](https://chakra-ui.com/). Even though we didn't have any experience with it, we decided to try it out since it was visualy appealing and popular. It is also very well documented and supported by the community.
+
+[Typescript](https://www.typescriptlang.org/) was a must have for us to ensure the quality of our code and to avoid bugs. Using a strongly typed codebase was also a good way to ensure the maintainability of our project in the future.
```

Comment:
```suggestion
[Typescript](https://www.typescriptlang.org/) was a must have for us to ensure the quality of our code and to avoid bugs. Using a strongly typed codebase was also a good way to ensure the maintainability of our project in the future.

The website sends requests to the client API with the use of Axios HTTP Client. Axios has several advantages as compared to the more well-known Fetch API. It is said to support more browsers, has built-in protection against XSRF (cross-site request forgery) and converts data to json format automatically. Axios is probably not the most feature-rich option present, but it is among the most straightforward and easy-to-use.
To simplify the development of the connection with API, our team have used the [OpenAPI Generator](https://openapi-generator.tech/) tool which generated typescript objects and functions for every controller and endpoint in our API. The autogenerated code uses Axios to send requests and is based on the OpenAPI specification of the API which was conveniently provided to us by NestJS once we added the `SwaggerModule`.

**Website structure**
The web application offers to users several public and protected routes, which give access to the main PaaSTech features. An anonymous user can view the home page, as well as pages for login, sign up, email verification and password reset, while an authenticated one have access to their personal dashboard which includes their profile, projects, project info, actions and logs, environment variables and project settings (which just allow to delete a project for now).
```